textile/features.textile

```diff
@@ -589,6 +589,7 @@ h3(#realtime-channel). RealtimeChannel
 ** @(RTL2d)@ A @ChannelStateChange@ object is emitted as the first argument for every @ChannelEvent@ (including both @RTL2a@ state changes and @RTL2g@ @UPDATE@ events). It may optionally contain a @reason@ consisting of an @ErrorInfo@ object; any state change triggered by a @ProtocolMessage@ that contains an @error@ member should populate the @reason@ with that error in the corresponding state change event
 ** @(RTL2f)@ When a channel @ATTACHED@ @ProtocolMessage@ is received, the @ProtocolMessage@ may contain a @RESUMED@ bit flag indicating that the channel has been resumed. The corresponding @ChannelStateChange@ (either @ATTACHED@ per @RTL2a@, or @UPDATE@ per @RTL12@) will contain a @resumed@ boolean attribute with value @true@ if the bit flag @RESUMED@ was included. When @resumed@ is @true@, this indicates that the channel attach resumed the channel state from an existing connection and there has been no loss of message continuity. In all other cases, @resumed@ is false. A test should exist to ensure that @resumed@ is always false when a channel first becomes @ATTACHED@, it is @true@ when the channel is @ATTACHED@ following a successful "connection recovery":#RTN16, and is @false@ when the channel is @ATTACHED@ following a failed "connection recovery":#RTN16
 ** @(RTL2h)@ Optionally, for backwards compatibility with 0.8 libraries, the @RealtimeChannel@ @EventEmitter@ can provide an overloaded method that supports @on(ChannelState)@, but must issue a deprecation warning
+** @(RTL2i)@ Optionally, when a channel @ATTACHED@ @ProtocolMessage@ is received, the @ProtocolMessage@ may contain a @HAS_BACKLOG@ bit flag indicating that the channel should expect a backlog of messages from a rewind or resume. The corresponding @ChannelStateChange@ may contain a @hasBacklog@ boolean attribute with value @true@ if the bit flag @HAS_BACKLOG@ was included.
```

Comment:
A couple of thoughts:

- Do we really need to make this optional? Ultimately, it's at the discretion of SDKs whether they implement _any_ of the spec points. It's a very minor thing to implement and seems like we should just say that SDKs should do it.
- The word "may" is a bit overloaded here — it's being used to describe a possible behaviour of the server and also an optional API choice from a client library. Makes it a bit harder to read than I think it needs to be.

Reply:
We use discretionary language like "optional" or "may" in plenty of other spec points so I don't see the harm tbh, especially for something like this where we have no plans to implement it in any SDK other than ably-js.

As for your second point, I've tweaked the wording slightly for readability but imo it's still necessary to highlight that both the bit flag and the api are optional. Let me know if you have any specific suggestions to improve it further

Reply:
OK, my opinions on making it optional aren't particularly strong.

As for wording, how about this (I'm not sure whether it's better or not):

> `ChannelStateChange` may optionally expose a boolean `hasBacklog` property. This property should be set to `true` if and only if the state change corresponds to an `ATTACHED` `ProtocolMessage` containing a `HAS_BACKLOG` bit flag.

Reply:
cool, i've changed it to that